analyses/long-format-table-utils/README.md

```diff
@@ -0,0 +1,84 @@
+
+- [Long-format table utils](#long-format-table-utils)
+  - [Purpose](#purpose)
+  - [Methods](#methods)
+    - [Update downloaded data that are used in this module](#update-downloaded-data-that-are-used-in-this-module)
+    - [Add gene and `cancer_group` annotations](#add-gene-and-cancer_group-annotations)
+      - [Implementation of long-format table annotator](#implementation-of-long-format-table-annotator)
+      - [API usage of long-format table annotator](#api-usage-of-long-format-table-annotator)
+      - [CLI usage of long-format table annotator](#cli-usage-of-long-format-table-annotator)
+
+### Purpose
+
+Create application programming interface (API) and command line interface (CLI) for handling [long-format tables](https://en.wikipedia.org/wiki/Wide_and_narrow_data#Narrow) that are generated by analysis modules. API provides analysis module developers with functions that can be imported into their own scripts via R `source('path/to/the/function/file.R')` or Python `import os, sys; sys.path.append(os.path.abspath("path/to/the/function/dir")); import function_filename_but_no_dot_py`. CLI provides analysis module developers with scripts that can be executed in their own run-module shell script with either `Rscript --vanilla path/to/the/script.R arg long.tsv long_edited.tsv` or `python3 path/to/the/script.py arg long.tsv long_edited.tsv`.
+
+This module is suggested by @jharenza and @kgaonkar6 in Slack at <https://opentargetspediatrics.slack.com/archives/C021Z53SK98/p1626290031138100?thread_ts=1626287625.133600&cid=C021Z53SK98>, in order to alleviate the burdens of analysis module developers for adding annotations and keeping track of what annotations need to be added. This module could also potentially handle large file storage issues at a later point, since the file size limit of GitHub is 100MB.
+
+| Sub-module name                                               | Implemented function                    | Available interface(s)                                                                                 |
+|---------------------------------------------------------------|-----------------------------------------|--------------------------------------------------------------------------------------------------------|
+| [`annotator`](#implementation-of-long-format-table-annotator) | Add gene and `cancer_group` annotations | [R API](#api-usage-of-long-format-table-annotator), [R CLI](#cli-usage-of-long-format-table-annotator) |
+
+API and CLI usages and descriptions are in the Methods section.
+
+### Methods
+
+#### Update downloaded data that are used in this module
+
+Run the following command to update downloaded data that are used in this module.
+
+```bash
+bash update-long-format-table-utils.sh
+```
+
+The `update-long-format-table-utils.sh` runs data downloading scripts in sub-modules, e.g. `annotator/run-download-annotation-data.sh`. The data downloading scripts use `git diff --stat` to check for data file changes.
```

Comment:
```suggestion
bash run-update-long-format-table-utils.sh


The `run-update-long-format-table-utils.sh` runs data downloading scripts in sub-modules, e.g. `annotator/run-download-annotation-data.sh`. The data downloading scripts use `git diff --stat` to check for data file changes.
```

I would suggest also starting the bash script with `run`, to keep convention with the other run scripts.

Reply:
Sounds good. I will add `run-` prefix to `update-long-format-table-utils.sh` where necessary.

Reply:
This suggestion is implemented in commit <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/55/commits/090f0808e5149dcf814a2dc7622d3dc45cafabf5>.